pkgs/development/python-modules/pytest-ruff/default.nix

```diff
@@ -0,0 +1,57 @@
+{
+  lib,
+  attrs,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  # build-system
+  poetry-core,
+  poetry-dynamic-versioning,
+
+  # dependencies
+  pytest,
+  ruff,
+
+  # check
+  pytestCheckHook,
+  pytest-mock,
+}:
+
+buildPythonPackage rec {
+  pname = "pytest-ruff";
+  version = "0.4.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "businho";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-Ol+W5mDGMCwptuBa0b+Plkm64UUBf9bmr9YBo8g93Ok=";
+  };
+
+  build-system = [
+    poetry-core
+    poetry-dynamic-versioning
+  ];
+
+  dependencies = [
+    pytest
+    ruff
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+  checkInputs = [
+    pytest-mock
+  ];
```

Comment:
```suggestion
  nativeCheckInputs = [
    pytestCheckHook
    pytest-mock
  ];
```

Reply:
wait why that? pytest-mock is platform independent.